engine/classes/Elgg/Di/ServiceProvider.php

```diff
@@ -524,4 +525,102 @@ public function __construct(Config $config) {
 		$this->setClassName('widgets', \Elgg\WidgetsService::class);
 	}
 
+	/**
+	 * Validate, normalize, fill in missing values, and lock some
+	 *
+	 * @param Config          $config Config
+	 * @param ServiceProvider $sp     Service Provider
+	 *
+	 * @return void
+	 * @throws ConfigurationException
+	 */
+	public function initConfig(Config $config, ServiceProvider $sp) {
```

Comment:
couldn't `$sp` be dropped and simply use `$this`?

Reply:
No, needs to be deffered